enhancements/ingress/optional-ingress-hypershift.md

```diff
@@ -0,0 +1,317 @@
+  - [GatewayClass](https://github.com/openshift/cluster-ingress-operator/blob/master/assets/gateway-api/gateway.networking.k8s.io_gatewayclasses.yaml)
+  - [Gateway](https://github.com/openshift/cluster-ingress-operator/blob/master/assets/gateway-api/gateway.networking.k8s.io_gateways.yaml)
+  - [HTTPRoute](https://github.com/openshift/cluster-ingress-operator/blob/master/assets/gateway-api/gateway.networking.k8s.io_httproutes.yaml)
+
+#### Related APIs
+
+The following APIs are left intact (see [Add Route API to the ingress capability](#add-route-api-to-the-ingress-capability) to know why) if the ingress capability is disabled:
+- [Route API](https://docs.openshift.com/container-platform/4.13/rest_api/network_apis/route-route-openshift-io-v1.html). Implemented in two places: OpenShift API as [CRD](https://github.com/openshift/api/blob/master/route/v1/route.crd.yaml) and in [openshift-api-server](https://github.com/openshift/openshift-apiserver/tree/master/pkg/route).
+- [Ingress Config API](https://docs.openshift.com/container-platform/4.13/rest_api/config_apis/ingress-config-openshift-io-v1.html). Implemented in OpenShift API as [CRD](https://github.com/openshift/api/blob/master/config/v1/0000_10_config-operator_01_ingress.crd.yaml).
+- [Ingress API](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#ingress-v1-networking-k8s-io). Implemented in Kubernetes.
+
+However the implementation of these APIs (or parts of them) depend on the ingress operator:
+- The ingress operator creates the default ingress controller to manage the backends for the routes. See [the drawbacks](#drawbacks) for further implications.
+  - The ingress objects are implicitly managed by the default ingress controller. See [the ingress to route controller risk](#ingress-to-route-controller) for further implications.
+- For [the component routes](https://docs.openshift.com/container-platform/4.13/rest_api/config_apis/ingress-config-openshift-io-v1.html#spec-componentroutes) the ingress operator creates the required role and role binding to get the custom TLS certificate secret from the `openshift-config` namespace. See [the component routes risk](#component-routes) for further implications.
+
+#### Console operator
+
+In case the ingress capability is not enabled in `ClusterVersion`, the following issues arise with the console components:
+- The console operator becomes degraded due to the failed [console route check](https://github.com/openshift/console-operator/blob/master/pkg/console/controllers/healthcheck/controller.go#L122).
+- The console route is not served by the default ingress controller.
+
+To resolve the former problem, it is necessary to skip [the console route check](https://github.com/openshift/console-operator/blob/master/pkg/console/controllers/healthcheck/controller.go#L122) when the ingress capability is not enabled.
+The later issue is discussed as a general case of the missing default ingress controller in [the drawbacks](#drawbacks).
```

Comment:
Another issue I can see from the console-operator's codebase is that the operator can get degraded if the `console` route wont get admitted. The thing is that we are using the [IngressURI](https://github.com/openshift/library-go/blob/master/pkg/route/routeapihelpers/routeapihelpers.go#L18) in some of the controllers, which is checking if the route has been admitted or not.

Reply:
The console operator chapter is [updated](https://github.com/openshift/enhancements/commit/a739c23e48fca39cdf126eb23e3d06fcc1b14304) to mention all the route admission checks which use `IngressURI`.

I believe that was the last point to detail. After the discussion we clarified [the Ingress Config API's point](https://github.com/openshift/enhancements/pull/1415#discussion_r1228166058): it won't break the console operator as this API stays intact.